src/ui/preview/PreviewHeader.vue

```diff
@@ -111,19 +107,21 @@ export default {
     };
   },
   watch: {
-    actionCollection(actionCollection) {
-      if (this.actionCollection) {
-        this.unlistenToActionCollection();
-      }
+    currentView() {
+      // wait for view to render with next tick
+      this.$nextTick(() => {
+        if (this.actionCollection) {
+          this.unlistenToActionCollection();
+        }
 
-      this.actionCollection.on('update', this.updateActionItems);
-      this.updateActionItems(this.actionCollection.getActionsObject());
-    }
-  },
-  mounted() {
-    if (this.actionCollection) {
-      this.actionCollection.on('update', this.updateActionItems);
-      this.updateActionItems(this.actionCollection.getActionsObject());
+        this.actionCollection = this.openmct.actions.getActionsCollection(
+          toRaw(this.objectPath),
+          toRaw(this.currentView)
+        );
```

Comment:
Interesting. I notice that strange things happen when we remove the `toRaw()`. I also noticed there are a few other places that don't use `toRaw()` before calling `getActionsCollection()`. I suggest we go with one of these solutions:

1. Update the other `getActionsCollection()` calls to use `toRaw()` on the reactive args passed in.
2. Implement a composable `useActionsCollection` that handles this logic and update the components to use this in its place.

Reply:
This is only happening here because the `this.currentView` isn't using the injected value, and was instead getting wrapped in a proxy because it's passed as a prop to this component. Everywhere else in Open MCT (except for here and `ObjectView`), we're using the injected value.

The proxy breaks `getActionsCollection()` because it uses the object as a key to lookup a requested `ActionsCollection`.

To make a long story short, I think it'd be overkill to update the rest of the codebase to handle this weird outlier 😀

Reply:
Ahhh ok, makes sense!